bundles/org.openhab.binding.flicbutton/src/main/java/org/openhab/binding/flicbutton/handler/FlicButtonHandler.java

```diff
@@ -0,0 +1,177 @@
+import io.flic.fliclib.javaclient.enums.DisconnectReason;
+
+/**
+ * The {@link FlicButtonHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Patrick Fink - Initial contribution
+ */
+public class FlicButtonHandler extends ChildThingHandler<FlicDaemonBridgeHandler> {
+
+    private Logger logger = LoggerFactory.getLogger(FlicButtonHandler.class);
+    private ScheduledFuture<?> delayedDisconnectTask;
+    private DisconnectReason latestDisconnectReason;
+    private ButtonConnectionChannel eventConnection;
+    private Bdaddr bdaddr;
+    BatteryStatusListener batteryConnection;
+
+    public FlicButtonHandler(Thing thing) {
+        super(thing);
+    }
+
+    public Bdaddr getBdaddr() {
+        return bdaddr;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // Pure sensor -> no commands have to be handled
+    }
+
+    @Override
+    public void initialize() {
+        super.initialize();
+        bdaddr = new Bdaddr((String) this.getThing().getConfiguration().get(CONFIG_ADDRESS));
+        if (bridgeValid) {
+            initializeThing();
+        }
+    }
+
+    public void initializeThing() {
+        try {
+            FlicButtonBatteryLevelListener batteryListener = new FlicButtonBatteryLevelListener(this);
+            BatteryStatusListener batteryConnection = new BatteryStatusListener(getBdaddr(), batteryListener);
+            bridgeHandler.getFlicClient().addBatteryStatusListener(batteryConnection);
+            this.batteryConnection = batteryConnection;
+
+            FlicButtonEventListener eventListener = new FlicButtonEventListener(this);
+            synchronized (eventListener) {
+                ButtonConnectionChannel eventConnection = new ButtonConnectionChannel(getBdaddr(), eventListener);
+                bridgeHandler.getFlicClient().addConnectionChannel(eventConnection);
+                this.eventConnection = eventConnection;
+                eventListener.wait(5000);
+                // Listener calls initializeStatus() before notifying so that ThingStatus is set at this point
+            }
+        } catch (IOException | InterruptedException e) {
+            logger.info("Connection setup for Flic Button {} failed.", this.getThing(), e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "Connection setup failed");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        cancelDelayedDisconnectTask();
+
+        try {
+            if (eventConnection != null) {
+                bridgeHandler.getFlicClient().removeConnectionChannel(eventConnection);
+            }
+            if (batteryConnection != null) {
+                bridgeHandler.getFlicClient().removeBatteryStatusListener(this.batteryConnection);
+            }
+        } catch (IOException e) {
+            logger.error("Error occured while removing button channel", e);
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be debug. See https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Under normal conditions, this is very unlikely to happen and could be an important hint for something being wrong. I decreased the log level to WARN.